lib/imapSimple.js

```diff
@@ -225,9 +261,49 @@ function connect(options, callback) {
     });
 }
 
+/**
+ * Returns a list of the parts.
+ * A part is either a message or an attachment.
+ * Parts are usually nested and numbered things like 1, 1.1, 1.2, ...
+ * Code taken from http://stackoverflow.com/questions/25247207/how-to-read-and-save-attachments-using-node-imap
+ */
+function findParts(struct, parts) {
+    parts = parts ||  [];
+    for (var i = 0; i < struct.length; i++) {
+        if (Array.isArray(struct[i])) {
+	    findParts(struct[i], parts);
+	} else if (struct[i].partID) {
+	    parts.push(struct[i]);
+	}
+    }
+    return parts;
+}
+
+/**
+ * Simplify the structure of the header (subject, date, etc) a bit
+ * Does some simple parsing to pull out useful attributes and all attachments
+ */
+function decorateMessage(message) {
+    if (message.parts.length != 1)
+        throw Error('Message has weird number of parts')
+    message._header = {};
+    ['from', 'to', 'subject', 'date'].forEach(function(key) {
```

Comment:
After considering it, I'm not really sure we'd want to include this method in the library. 

There are only a couple of the header properties where only one value would be the norm. In the case of the 4 values that are being extracted here, `from` and `to` could potentially include multiple values, so grabbing just the first may not really be all that useful. `subject` and `date` are kind of nice to be able to just access without array notation, but since all the other headers are accessed with array notation, it seems consistent to just require it for all the headers. 

I think a method like this is probably best left in user code.